src/gbxml/MapSchedules.cpp

```diff
@@ -218,7 +218,7 @@ namespace gbxml {
       OS_ASSERT(yd.calendarYear().get() == std::stoi(endDateParts.at(0)));
       openstudio::Date endDate = yd.makeDate(std::stoi(endDateParts.at(1)), std::stoi(endDateParts.at(2)));
 
-      std::string weekScheduleId = element.child("WeekScheduleId").attribute("weekScheduleIdRef").value();
+      std::string weekScheduleId = scheduleYearElement.child("WeekScheduleId").attribute("weekScheduleIdRef").value();
```

Comment:
Actual fix...

Really we need error handling in gbxml... There is currently no validation against the gbXML schema happening and there is very little checking that is done manually (like ensuring that required values/attributes are supplied, or checking that a YearSchedule references a `weekScheduleIdRef` you make sure it can be found)

I'll open a separate enhancement request for that.

Reply:
indeed, it's worth discussing adding in some logic or external lib that can validate gbXML.  Thanks for opening up an enhancement request.